scripts/tests/py/metadata.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/local/bin/python3
+# Copyright (c) 2022 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+import re
+import yaml
+from typing import Optional
+from typing import Dict
+from typing import List
+from typing import Union
+
+@dataclass
+class Metadata:
+    py_script_path: Optional[str] = None
+    run: Optional[str] = None
+    app: Optional[str] = None
+    factoryreset: Optional[bool] = False
+    comissioning_method: Optional[str] = None
+    discriminator: Optional[int] = None
+    kvs: Optional[str] = None
+    storage_path: Optional[str] = None
+    on_network_commission: Optional[str] = None
+    passcode: Optional[int] = None
+    endpoint: Optional[str] = None
+    manual_code: Optional[str] = None
+    PICS: Optional[str] = None
+    tests: Optional[str] = None
+
+
+class Metadata_Reader:
```

Comment:
```suggestion
class MetadataReader:
```

Maybe as naming, I think we do not often use underscores and CamelCase is sufficient